core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2342,6 +2342,12 @@ private[spark] object Utils extends Logging {
    * Return the initial number of executors for dynamic allocation.
    */
   def getDynamicAllocationInitialExecutors(conf: SparkConf): Int = {
+    if (conf.get(DYN_ALLOCATION_INITIAL_EXECUTORS) < conf.get(DYN_ALLOCATION_MIN_EXECUTORS)) {
+      logWarning(s"${DYN_ALLOCATION_INITIAL_EXECUTORS.key} less than " +
+        s"${DYN_ALLOCATION_MIN_EXECUTORS.key} is invalid, will use " +
+          s"${DYN_ALLOCATION_MIN_EXECUTORS} instead.")
```

Comment:
also note that the comment above isn't exactly correct because if the number of executor instances is > then the min it would use that number instead of the min executors.